### DISCUSSION TITLE
프로젝트 회고

### DISCUSSION THREAD

**Discussion by DaleSeo** [DaleStudy/leaderboard#93] [Announcements]
함께 프로젝트를 하면서 좋았던 점, 아쉬웠던 점들을 댓글로 달아주세요 :)

**Comment by SamTheKorean** [comment]
좋았던 점 : 회사에서는 기한에 쫒기며 개발할 때가 많았지만 프로젝트에선 성숙한 개발문화를 지향하며 경험에서 좋았습니다.

**Comment by SamTheKorean** [comment]
좋았던 점 : 팀원들의 꼼꼼하고 건설적인 피드백을 받으면 다시 한 번 생각해볼 수 있어서 많이 배웠습니다.

**Comment by yolophg** [comment]
좋았던 점 : 디자이너로서 프로젝트에 참여해본 것은 처음이라, 굉장히 챌린지인 동시에 디자이너의 관점을 이해하고 프론트엔드 개발자로서의 시야를 넓힐 수 있어 새롭고 좋은 경험이었다. 프로덕트 디자이너들의 컨트롤 범위가 머리로 알던 것과 몸소 경험하니 단순히 디자인을 뛰어넘어 훨씬 더 광범위했고, 그들의 위대함을 느낄 수 있었다.

**Comment by sounmind** [comment]
## 좋았던 점
- 다양한 배경을 가진 개발자분들과 협업할 수 있었다.
- 자발적으로 페어 프로그래밍을 여러 번 해봤는데 서로 다른 관점을 공유하는 능력을 키우는 계기가 되었다.
- Vitest, CI, CD, 스토리북 등 React 말고도 여러 기술들을 배울 수 있었다.

## 아쉬웠던 점
- 막판에 일이 바빠져서 절대적인 시간을 더 투자하지 못해서 아쉬웠다.
- 여기서 배운 것들을 정리하거나 개인 프로젝트에 적용해보지 못해서 아쉽다. 달레님이 준 링크, 소스들을 다시 읽어봐야겠다.

**Comment by yolophg** [comment]
아쉬웠던 점: 사실상 디자이너 롤에 인원이 없어서 개발자를 생각했지만 갑자기 디자이너라는 책임을 안게 되어, 공부하고 적용할 시간이 절대적으로 부족했고 이로 인해 극 초반에는 이게 병목현상을 일으키기도 했다. 이번 리더보드 프로덕트 스펙 대비 디자이너 경험이 없는 내가 맡기에는 디자이너로서의 기준과 확신이 부족했다. 
따라서, 디자이너 롤을 원하는 팀원 or 프로덕트 디자이너 경험이 있는 팀원을 모집하는 것이 프로덕트에도 효율적이고 좋은 선택이 될 것 같다고 생각했다.

**Comment by yolophg** [comment]
좋았던 점: 모든 팀원들이 스프린트 기한에 맞춰 열심히 하려 노력했고, 심지어 스프린트가 연장 되었음에도 불구하고 단 한 명의 하차도 없이 끝까지 힘을 합쳐 프로젝트 마지막까지 이끌었다.

**Comment by yolophg** [comment]
좋았던 점: a11y에 대해 또 한 번 배울 수 있었다. 모던하고 디자이너로서 추구하는 디자인과 동시에 a11y 기준을 우선시로 고려해야하는 디자인 사이에서 여러 번 a11y 에러를 경험할 수 밖에 없었고 이 간극을 맞춰가는 것이 꽤나 어려우면서도 흥미로운 챌린지였다.

**Comment by yolophg** [comment]
좋았던 점: Github, storybook, CI/CD를 더욱 잘 활용하여 스프린트 및 프로젝트, 각 PR에서 이뤄지는 UI 테스트 등을  직관적이고 정갈하게 관리할 수 있어 좋았다. 또, Chromatic을 이 프로젝트에서 경험해볼 수 있어서 아주 유익했다.

**Comment by SamTheKorean** [comment]
좋았던 점 : 스토리 북 등등 새로운 기술들을 배울 수 있어 좋았습니다.

**Comment by Sunjae95** [comment]
## 좋았던점
1. test와 a11y등을 처음 접해보는 기술을 고려한 프로젝트를 경험할 수 있어서 좋았습니다. 이전부터 해보고싶었지만 어떻게 헤야하지 몰라서 못하고 있었는데 이렇게 사이드 프로젝트를 통해 경험하게 되어 만족스럽습니다.
2. 깃허브 프로젝트, 이슈등 깃허브로 협업의 경험이 좋았습니다. 깃허브로 협업 가능성을 알게 되고 한곳에서 관리하니 작업의 집중도가 높았던거 같았습니다.
3. 상호간의 리뷰를 통해 개인적으로나 프로젝트가 성장하는것을 느꼈습니다. 서로 코드를 공유하고 조금 더 명시적이고 불필요한 코드는 제거하는 등 과정이 모두의 성장에 기여하지 않았나 생각합니다.
4. 서로의 생각을 존중해주고 더 나은길을 찾아가는 과정이 좋았습니다.

## 아쉬웠던점
1. 모든 구성원의 코드리뷰를 진행하지 못했던점. 시간을 많이 할애하고 싶었지만 그러지 못해서 팀원들의 코드를 모두 볼 수가 없었다.
2. 취업후 출근으로 인해 10주차부터 프로젝트에 참여를 많이 못했던점. 일하면서 하시는 분들 존경합니다..🥲
3. 새롭게 도전하는 기술들이 많아서 모든 기술을 검토해보는것 보단 팀원에게 받는 경험만 해서 아쉬웠습니다. 내가 알았더라면 서로 더 많은 성장을 할 수 있지 않았을까? 라는 생각이 스쳐지나 갑니다..ㅎㅎ
4. 프로젝트의 완성도는 높아졌지만 기한이 계속 늘어나서 아쉬웠습니다. 하다보니 언제끝날 수 있을까?라는 생각을 하는 제 모습을 발견해서 아쉬웠습니다.

**Reply by Sunjae95** [reply]
모두들 함께해주셔서 감사합니다! 🙂

**Comment by yolophg** [comment]
아쉬웠던 점: 중간 회고가 부족해서 아쉬웠다.

**Comment by HC-kang** [comment]
### 좋았던 점

- 번과 크로매틱, 프론트엔드 테스트 등을 처음으로 접해서, 새로운 개발 도구를 많이 접할 수 있어서 좋았습니다.
- 잘 몰랐던 깃허브의 프로젝트 관리 기능을 배울 수 있었습니다.

### 아쉬웠던 점

- 어느 순간 업무로 인해 참여도가 줄어들고, 특히 프론트엔드 개발에 참여하지 못해서 아쉽습니다.
- 초기에는 페어 프로그래밍도 진행했지만, 위 이유로 인해 프로젝트 후반에는 페어를 할 기회를 잃어서 아쉽습니다.
